src/components/Reports/GoalCalculator/SummaryReport/SummaryReport.test.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect } from 'react';
+import { ThemeProvider } from '@mui/material/styles';
+import { render } from '@testing-library/react';
+import { SnackbarProvider } from 'notistack';
+import TestRouter from '__tests__/util/TestRouter';
+import { GqlMockedProvider } from '__tests__/util/graphqlMocking';
+import {
+  afterTestResizeObserver,
+  beforeTestResizeObserver,
+} from '__tests__/util/windowResizeObserver';
+import theme from 'src/theme';
+import {
+  GoalCalculatorReportEnum,
+  GoalCalculatorStepEnum,
+} from '../GoalCalculatorHelper';
+import {
+  GoalCalculatorProvider,
+  useGoalCalculator,
+} from '../Shared/GoalCalculatorContext';
+import { SummaryReport } from './SummaryReport';
+
+interface ContextHelperProps {
+  selectedReport: GoalCalculatorReportEnum;
+}
+
+const ContextHelper: React.FC<ContextHelperProps> = ({ selectedReport }) => {
+  const { handleStepChange, setSelectedReport } = useGoalCalculator();
+
+  useEffect(() => {
+    handleStepChange(GoalCalculatorStepEnum.SummaryReport);
+    setSelectedReport(selectedReport);
+  }, []);
+
+  return null;
+};
```

Comment:
instead of re-creating the context, why don't you use the real context and update the value?
Like we do on other tests.
```
UserPreferenceContext.Provider value={{ locale: 'en-US' }}>
```

Reply:
Whenever possible, I like to use real providers instead of mocking them to ensure that we're testing the actual behavior of the app. I modified `ContextHelper` to add buttons that change the report like we've done in some other tests. Let me know if that's OK.